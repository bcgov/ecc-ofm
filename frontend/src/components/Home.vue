<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <v-card
    title="OFM"
    subtitle="Temp Home Screen"
    text="..."
    style="height:100%;"
  ></v-card>
</template>
