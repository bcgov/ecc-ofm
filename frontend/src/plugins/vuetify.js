import '@fortawesome/fontawesome-free/css/all.css';
import '@mdi/font/css/materialdesignicons.css';
import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import { Ripple } from 'vuetify/lib/directives';

Vue.use(Vuetify, {
  directives: {
    Ripple
  }
});

export default new Vuetify({
  theme: {
    light: true,
    dark: false
  },
  icons: {
    iconfont: 'fa',
    values: {
      login: 'fas fa-user-clock',
      fast: 'fas fa-shipping-fast',
      sign_in: 'fas fa-sign-in-alt',
      info1: 'fas fa-info-circle',
      downArrow: 'fas fa-angle-down',
      upArrow: 'fas fa-angle-up',
      user: 'far fa-user',
      copy: 'fas fa-copy',
      search: 'fas fa-search',
      error: 'fas fa-exclamation-triangle',
      lock: 'fas fa-lock',
      info2: 'fas fa-info-circle fa-10x',
      question: 'fas fa-question-circle fa-10x',
      plus: 'fas fa-plus-circle',
      minus: 'fas fa-minus-circle',
    }
  }
});
