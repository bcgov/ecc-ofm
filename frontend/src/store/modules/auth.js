import ApiService from '@/common/apiService';
/* TODO: uncomment during integration with authservice...
import AuthService from '@/common/authService';
*/
import { Routes } from '@/utils/constants';
import {defineStore} from 'pinia';

export const authStore = defineStore('auth', {
  namespaced: true,
  state: () => ({
    acronyms: [],
    isAuthenticated: true  /* TODO: during integration with backend, replace true with... localStorage.getItem('jwtToken') !== null */,
    isAuthorizedUser: true /* TODO: during integration with backend, replace true with... localStorage.getItem('isAuthorizedUser') !== null */,
    userInfo: false,
    isValidChildCareProviderUser: localStorage.getItem('iisValidChildCareProviderUser') !== null,
    isValdProgramUser: localStorage.getItem('isValdProgramUser') !== null,
    isValidFinancialOpsUser: localStorage.getItem('isValidFinancialOpsUser') !== null,    
  }),
  getters: {
    acronymsGet: state => state.acronyms,
    isAuthenticatedGet: state => state.isAuthenticated,
    jwtTokenGet: () => localStorage.getItem('jwtToken'),
    userInfoGet: state => state.userInfo,
    //TODO: 3 temp roles ('CCP_ROLE', 'OPS_ROLE', 'PCM_ROLE') were created in auth.js (loosely
    //based on OFM requirements) for the purpose of achieving a 1st draft of the frontend that
    //will render a home screen and menu with minimal errors given no authorization/backend integration.
    //Thus the following related code is only temporarly and expected to be replaced...
    CCP_ROLE: state => state.isValidChildCareProviderUser,
    OPS_ROLE: state => state.isValdProgramUser,
    PCM_ROLE: state => state.isValidFinancialOpsUser
  },
  actions: {
    //sets Json web token and determines whether user is authenticated
    async setJwtToken(token = null) {
      if (token) {
        this.isAuthenticated = true;
        localStorage.setItem('jwtToken', token);
      } else {
        this.isAuthenticated = false;
        localStorage.removeItem('jwtToken');
      }
    },
    async setAuthorizedUser(isAdminUser){
      if (isAdminUser) {
        this.isAuthorizedUser = true;
        localStorage.setItem('isAuthorizedUser', 'true');
      } else {
        this.isAuthorizedUser = false;
        localStorage.removeItem(('isAuthorizedUser'));
      }
    },
    async setAuthorizedWebsocketUser(isAuthorizedWebsocketUser){
      if (isAuthorizedWebsocketUser) {
        this.isAuthorizedWebsocketUser = true;
        localStorage.setItem('isAuthorizedWebsocketUser', 'true');
      } else {
        this.isAuthorizedWebsocketUser = false;
        localStorage.removeItem(('isAuthorizedWebsocketUser'));
      }
    },
    async setUserInfo(userInf) {
      if (userInf) {
        this.userInfo = userInf;
      } else {
        this.userInfo = null;
      }
    },
    // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setChildCareProviderUser(isValidChildCareProviderUser) {
      if (isValidChildCareProviderUser) {
        this.isValidChildCareProviderUser = true;
        localStorage.setItem('isValidChildCareProviderUser', 'true');
      } else {
        this.isValidChildCareProviderUser = false;
        localStorage.removeItem(('isValidChildCareProviderUser'));
      }
    },
    // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setProgramUser(isValdProgramUser) {
      if (isValdProgramUser) {
        this.isValdProgramUser = true;
        localStorage.setItem('isValdProgramUser', 'true');
      } else {
        this.isValdProgramUser = false;
        localStorage.removeItem(('isValdProgramUser'));
      }
    },
      // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setFinancialOpsUser(isValidFinancialOpsUser) {
      if (isValidFinancialOpsUser) {
        this.isValidFinancialOpsUser = true;
        localStorage.setItem('isValidFinancialOpsUser', 'true');
      } else {
        this.isValidFinancialOpsUser = false;
        localStorage.removeItem(('isValidFinancialOpsUser'));
      }
    },
    //sets the token required for refreshing expired json web tokens
    async logoutState() {
      localStorage.removeItem('jwtToken');
      this.userInfo = false;
      this.isAuthenticated = false;
    },
    async getUserInfo() {
      const token = localStorage.getItem('jwtToken');

      if(!token){
        await this.getJwtToken();
      }

      if(localStorage.getItem('jwtToken')) {
        const response = await ApiService.apiAxios.get(Routes.USER);
        await this.setUserInfo(response.data);
      }
    },
    //retrieves the json web token from local storage. If not in local storage, retrieves it from API
    async getJwtToken() {
      try {
        const token = localStorage.getItem('jwtToken');
        if (this.isAuthenticated && !!token) {
          /* TODO: uncomment during integration with authservice/backend...
          const response = await AuthService.refreshAuthToken(token);
          */
         //TODO: uncomment 'response' below during integration with backend...
          await this.setAuthorizations(/*response*/);
        } else {
          /* TODO: uncomment during integration with backend...
          const response = await AuthService.getAuthToken();
          */
         //TODO: uncomment 'response' below during integration with backend...
          await this.setAuthorizations(/*response*/);
        }
      } catch (e) {
        // Remove tokens from localStorage and update state
        await this.setJwtToken(null);
        throw e;
      }
    },
    //TODO: uncomment 'response' below during integration with authorization/backend...
    async setAuthorizations(/*response*/) {
    /* TODO: uncomment during integration with backend...
      if (response.jwtFrontend) {
        await this.setJwtToken(response.jwtFrontend);
      }

      ApiService.setAuthHeader(response.jwtFrontend);
    */
      await this.setAuthorizedUser(true);
      await this.setChildCareProviderUser(true);
      await this.setProgramUser(true);
      await this.setFinancialOpsUser(true);
    }
  }
});
