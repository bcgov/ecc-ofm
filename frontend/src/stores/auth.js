import moment from 'moment'
import { defineStore } from 'pinia'

import ApiService from '@/common/apiService'
import AuthService from '@/common/authService'
import { useAppStore } from '@/stores/app'
import { APPLICATION_INTAKE_TYPES, OFM_PROGRAM_CODES, ROLES } from '@/utils/constants'

export const useAuthStore = defineStore('auth', {
  namespaced: true,
  state: () => ({
    isAuthenticated: localStorage.getItem('jwtToken') !== null,
    isMinistryUser: false,
    isImpersonating: false,
    isUserInfoLoaded: false,
    userInfo: null,
    impersonateId: null,
    currentFacility: {},
    permissions: [],
  }),
  getters: {
    isActingProvider: (state) => !state.isMinistryUser || state.isImpersonating,
    hasFacilities: (state) => state.userInfo?.facilities?.length > 0,
    hasPermission: (state) => {
      return (permission) => {
        return state.permissions?.some((p) => p === permission)
      }
    },
  },
  actions: {
    //sets Json web token and determines whether user is authenticated
    async setJwtToken(token = null) {
      if (token) {
        this.isAuthenticated = true
        localStorage.setItem('jwtToken', token)
      } else {
        this.isAuthenticated = false
        localStorage.removeItem('jwtToken')
      }
    },
    //sets the token required for refreshing expired json web tokens
    async logout() {
      localStorage.removeItem('jwtToken')
      this.userInfo = null
      this.isAuthenticated = false
    },
    async getUserInfo() {
      const token = localStorage.getItem('jwtToken')
      if (!token) {
        await this.getJwtToken()
      }
      if (localStorage.getItem('jwtToken')) {
        if (!this.isUserInfoLoaded) {
          const appStore = useAppStore()
          let userInfoRes = undefined
          if (this.impersonateId && this.isMinistryUser) {
            userInfoRes = await ApiService.getUserImpersonateInfo(this.impersonateId)
            this.isImpersonating = true
          } else {
            userInfoRes = await ApiService.getUserInfo()
            this.isMinistryUser = userInfoRes.data.isMinistryUser
            delete userInfoRes.data.isMinistryUser
          }
          this.userInfo = userInfoRes.data
          // Default the facility
          if (this.isActingProvider && this.hasFacilities) {
            this.currentFacility = this.userInfo.facilities[0]
          }
          
          // Lookup the permissions
          let role
          if (this.isImpersonating) {
            // When impersonating always use 'Impersonate', not the impersonated user's role
            role = appStore.roles.find((role) => role.roleName === ROLES.IMPERSONATE)
          } else {
            role = appStore.roles.find((role) => role.roleId === this.userInfo.role?.ofm_portal_roleid)
          }
          this.permissions = role?.permissions.map((p) => p.permissionName)

          /*
            A facility can apply for a core application if it satisfies these conditions:
            1. There is an Open Intake or the facility is in the allowed facility list of a Limited Intake
            2. If the facility is in the CCOF program, their program start date must >= 1 year from the current date.
          */
          this.userInfo?.facilities?.forEach((facility) => {
            facility.intakeWindowCheckForAddApplication = appStore.applicationIntakes?.some((intake) => {
              const isWithinApplicationIntakeWindow = moment().isSameOrAfter(moment(intake.startDate)) && moment().isSameOrBefore(moment(intake.endDate))
              const isOpenIntake = intake.type === APPLICATION_INTAKE_TYPES.OPEN_INTAKE
              const limitedIntakeFacilities = intake?.facilities?.map((facility) => facility.facilityId)
              return isWithinApplicationIntakeWindow && (isOpenIntake || limitedIntakeFacilities?.includes(facility.facilityId))
            })
            facility.ccofEnrolmentCheckForAddApplication = facility?.programCode !== OFM_PROGRAM_CODES.CCOF || (facility?.programCode === OFM_PROGRAM_CODES.CCOF && facility?.ccofOneYearEnrolment)
          })

          this.isUserInfoLoaded = true
        }
      }
    },
    //retrieves the json web token from local storage. If not in local storage, retrieves it from API
    async getJwtToken() {
      try {
        const token = localStorage.getItem('jwtToken')
        if (this.isAuthenticated && !!token) {
          const response = await AuthService.refreshAuthToken(token)
          await this.setAuthorizations(response)
        } else {
          const response = await AuthService.getAuthToken()
          await this.setAuthorizations(response)
        }
      } catch (e) {
        // Remove tokens from localStorage and update state
        await this.setJwtToken(null)
        throw e
      }
    },
    async setAuthorizations(response) {
      if (response.jwtFrontend) {
        await this.setJwtToken(response.jwtFrontend)
      }
      ApiService.setAuthHeader(response.jwtFrontend)
    },
  },
})
