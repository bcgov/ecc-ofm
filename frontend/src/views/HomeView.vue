<template>
  <AppHeroImage />
  <OrganizationHeader />
  <v-container v-bind="$attrs">
    <v-row>
      <v-col class="pa-1">
        <p class="home-overview page-overview">Welcome to your Management Portal - Connect with the program, view and manage accounts, applications, and reports</p>
      </v-col>
    </v-row>
    <v-row v-if="loading">
      <v-col v-for="n in 6" :key="n" cols="12" md="6" lg="4">
        <v-skeleton-loader :loading="loading" type="card" />
      </v-col>
    </v-row>
    <v-row v-else>
      <v-col v-if="hasPermission(PERMISSIONS.SEARCH_VIEW_REPORTS)" cols="12" md="6" lg="4">
        <v-card class="basic-card" id="reporting-card" prepend-icon="mdi-file-chart-outline" title="Reporting" @click="$router.push({ name: 'reporting' })">
          <v-card-text>Complete or view current or past Monthly Reports and submit financial reports.</v-card-text>
        </v-card>
      </v-col>
      <v-col v-if="hasPermission([PERMISSIONS.VIEW_FUNDING_AGREEMENT, PERMISSIONS.VIEW_FUNDING_AMOUNTS])" cols="12" md="6" lg="4">
        <v-card class="basic-card" id="funding-card" prepend-icon="mdi-currency-usd" title="Funding" @click="$router.push({ name: 'funding-overview' })">
          <v-card-text>Review operational funding details by month or funding envelopes.</v-card-text>
        </v-card>
      </v-col>
      <v-col v-if="hasPermission([PERMISSIONS.MANAGE_NOTIFICATIONS])" cols="12" md="6" lg="4">
        <v-card class="basic-card" id="assistance-card" prepend-icon="mdi-message-text-outline" title="Assistance Request" @click="toggleAssistanceRequestDialog">
          <v-card-text>Have Questions? Send us a message.</v-card-text>
        </v-card>
      </v-col>
      <v-col v-if="hasPermission(PERMISSIONS.VIEW_APPLICATIONS)" cols="12" md="6" lg="4">
        <v-card class="basic-card" id="applications-card" prepend-icon="mdi-file-document-multiple-outline" title="Applications" @click="$router.push({ name: 'applications-history' })">
          <v-card-text>Submit new or view applications for $10 a Day funding or Allowances.</v-card-text>
        </v-card>
      </v-col>
      <v-col v-if="hasPermission(PERMISSIONS.VIEW_ORG_FACILITY, PERMISSIONS.MANAGE_USERS_VIEW)" cols="12" md="6" lg="4">
        <v-card class="basic-card" id="account-mgmt-card" prepend-icon="mdi-cog-outline" title="Account Management" @click="$router.push({ name: 'account-mgmt' })">
          <v-card-text>Maintain or edit organization or facility information and request a change.</v-card-text>
        </v-card>
      </v-col>
      <v-col cols="12" md="6" lg="4">
        <v-card class="basic-card" id="help-card" prepend-icon="mdi-help-circle-outline" title="Help and Resources" @click="$router.push({ name: 'help' })">
          <v-card-text>Need support? Find program training tools and resources, technical help, or call us.</v-card-text>
        </v-card>
      </v-col>
    </v-row>
  </v-container>

  <SignFundingPopup v-if="hasPermission(PERMISSIONS.VIEW_FUNDING_AGREEMENT)" @loading="setLoading" />
  <NewRequestDialog v-if="hasPermission(PERMISSIONS.MANAGE_NOTIFICATIONS)" :show="showAssistanceRequestDialog" @close="toggleAssistanceRequestDialog" />
</template>

<script>
import SignFundingPopup from '@/components/funding/SignFundingPopup.vue'
import NewRequestDialog from '@/components/messages/NewRequestDialog.vue'
import OrganizationHeader from '@/components/organizations/OrganizationHeader.vue'
import AppHeroImage from '@/components/ui/AppHeroImage.vue'
import permissionsMixin from '@/mixins/permissionsMixin.js'

export default {
  name: 'HomeView',
  components: { AppHeroImage, NewRequestDialog, OrganizationHeader, SignFundingPopup },
  mixins: [permissionsMixin],
  emits: ['setLoading'],
  data() {
    return {
      loading: false,
      showAssistanceRequestDialog: false,
    }
  },
  methods: {
    setLoading(value) {
      this.loading = value
    },
    toggleAssistanceRequestDialog() {
      this.showAssistanceRequestDialog = !this.showAssistanceRequestDialog
    },
  },
}
</script>

<style scoped>
.home-overview {
  text-align: center;
}
</style>
