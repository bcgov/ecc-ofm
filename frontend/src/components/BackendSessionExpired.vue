<template>
  <div style="display: none">
    <a id="logout_href" :href="link" />
  </div>
</template>

<script>
import { Routes } from '@/utils/constants'
import { useAuthStore } from '@/stores/auth'
import { mapActions } from 'pinia'
export default {
  name: 'BackendSessionExpired',
  data() {
    return {
      link: Routes.SESSION_EXPIRED
    }
  },
  mounted() {
    window.location = document.getElementById('logout_href').href
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    logout() {
      this.setJwtToken()
    }
  }
}
</script>
