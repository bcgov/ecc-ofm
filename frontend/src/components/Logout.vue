<template>
  <v-container fluid class="full-height">
    <!-- login article -->
    <article name="logout-banner">
      <v-row align="center" justify="center">
        <v-col xs="10" sm="10" md="8" lg="4" xl="3">
          <v-card class="session-expired-card">
            <v-card-title class="gov-header">
              <h4 id="logout_text">Logged Out</h4>
            </v-card-title>
            <v-card-text id="logout_descriptor">
              <v-row style="margin: 0.3rem"> You have Logged out. </v-row>
              <a
                id="login-button"
                :href="routes.LOGIN"
                class="ma-1"
                dark
                color="#003366"
                @click="clearStorage"
                >Log In</a
              ><span>again if you wish to continue.</span>
            </v-card-text>
          </v-card>
        </v-col>
      </v-row>
    </article>
  </v-container>
</template>

<script>
import { mapActions } from 'pinia'
import { Routes } from '../utils/constants'
import { useAuthStore } from '@/stores/auth'

export default {
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Logout',

  data() {
    return {
      routes: Routes
    }
  },
  created() {
    this.setJwtToken()
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    clearStorage() {
      this.setJwtToken()
    }
  }
}
</script>

<style scoped>
.full-height {
  height: 100%;
}
.session-expired-card {
  margin-top: 15rem;
  width: 100%;
  background: #d9e7d8;
}
</style>
