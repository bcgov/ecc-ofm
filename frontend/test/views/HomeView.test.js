import { describe, test } from 'vitest'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

import { useAuthStore } from '@/stores/auth.js'
import HomeView from '@/views/HomeView.vue'
import { createTestingPinia } from '@pinia/testing'
import { mount } from '@vue/test-utils'

const vuetify = createVuetify({ components, directives })

describe('HomeView', () => {
  const global = { global: { plugins: [createTestingPinia(), vuetify] } }
  const authStore = useAuthStore()
  authStore.isAuthenticated = true
  test('BceID User', async () => {
    authStore.userInfo = { userName: 'ofmqa01' }
    authStore.userInfo = { userName: 'ofmqa01', role: 2 }
    const wrapper = mount(HomeView, global)

    // BCeID User should see all options
    expect(wrapper.text()).toContain('Reporting')
    expect(wrapper.text()).toContain('Funding')
    expect(wrapper.text()).toContain('Documents')
    expect(wrapper.text()).toContain('View Applications')
    expect(wrapper.text()).toContain('New Application')
    expect(wrapper.text()).toContain('Resources')
    expect(wrapper.text()).toContain('Account Management')

    expect(wrapper.html()).toMatchSnapshot()
  })
  test('IDIR User', async () => {
    authStore.isMinistryUser = true
    authStore.userInfo = { userName: 'idiruser' }
    authStore.userInfo = { userName: 'idiruser', role: 2 }
    const wrapper = mount(HomeView, global)

    // IDIR User should see all options
    expect(wrapper.text()).toContain('Reporting')
    expect(wrapper.text()).toContain('Funding')
    expect(wrapper.text()).toContain('Documents')
    expect(wrapper.text()).toContain('View Applications')
    expect(wrapper.text()).toContain('New Application')
    expect(wrapper.text()).toContain('Resources')
    expect(wrapper.text()).toContain('Account Management')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
