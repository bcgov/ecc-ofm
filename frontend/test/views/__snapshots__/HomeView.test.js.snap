// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView > BceID User 1`] = `
"<div>
  <div class=\\"v-container v-locale--is-ltr\\">
    <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
      <!---->
      <div class=\\"v-card__loader\\">
        <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
          <!---->
          <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
          <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
            <div class=\\"v-progress-linear__indeterminate\\">
              <div class=\\"v-progress-linear__indeterminate long\\"></div>
              <div class=\\"v-progress-linear__indeterminate short\\"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div class=\\"v-card-title\\">Provider User</div>
      <div class=\\"v-card-text\\">
        <div class=\\"v-list v-theme--light v-list--density-default v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">Business BCeID:</div>
              <div class=\\"v-list-item-subtitle\\">ofmqa01</div>
            </div>
            <!---->
          </div>
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">GUID:</div>
              <div class=\\"v-list-item-subtitle\\"></div>
            </div>
            <!---->
          </div>
          <!--v-if-->
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">Name:</div>
              <div class=\\"v-list-item-subtitle\\"> </div>
            </div>
            <!---->
          </div>
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">Email:</div>
              <div class=\\"v-list-item-subtitle\\"></div>
            </div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <!----><span class=\\"v-card__underlay\\"></span>
    </div>
  </div>
  <div class=\\"v-container v-locale--is-ltr\\">
    <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
      <!---->
      <div class=\\"v-card__loader\\">
        <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
          <!---->
          <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
          <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
            <div class=\\"v-progress-linear__indeterminate\\">
              <div class=\\"v-progress-linear__indeterminate long\\"></div>
              <div class=\\"v-progress-linear__indeterminate short\\"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div class=\\"v-card-title\\">Roles</div>
      <div class=\\"v-card-text\\">
        <div class=\\"v-list v-theme--light v-list--density-default v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-subtitle\\"> (WIP: will convert #s to descriptions)</div>
            </div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <!----><span class=\\"v-card__underlay\\"></span>
    </div>
  </div>
  <div class=\\"v-container v-locale--is-ltr\\">
    <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
      <!---->
      <div class=\\"v-card__loader\\">
        <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
          <!---->
          <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
          <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
            <div class=\\"v-progress-linear__indeterminate\\">
              <div class=\\"v-progress-linear__indeterminate long\\"></div>
              <div class=\\"v-progress-linear__indeterminate short\\"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div class=\\"v-card-title\\">Organization</div>
      <div class=\\"v-card-text\\">
        <div class=\\"v-list v-theme--light v-list--density-default v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">ID:</div>
              <div class=\\"v-list-item-subtitle\\"></div>
            </div>
            <!---->
          </div>
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">Account Number:</div>
              <div class=\\"v-list-item-subtitle\\"></div>
            </div>
            <!---->
          </div>
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">Name:</div>
              <div class=\\"v-list-item-subtitle\\"></div>
            </div>
            <!---->
          </div>
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">Type:</div>
              <div class=\\"v-list-item-subtitle\\"> (WIP: will convert # to description)</div>
            </div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <!----><span class=\\"v-card__underlay\\"></span>
    </div>
  </div>
  <div class=\\"v-container v-locale--is-ltr\\">
    <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
      <!---->
      <div class=\\"v-card__loader\\">
        <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
          <!---->
          <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
          <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
            <div class=\\"v-progress-linear__indeterminate\\">
              <div class=\\"v-progress-linear__indeterminate long\\"></div>
              <div class=\\"v-progress-linear__indeterminate short\\"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div class=\\"v-card-title\\">Current Facility</div>
      <div class=\\"v-card-text\\"> Facility Name: <br><span style=\\"opacity: 0.6;\\"></span></div>
      <!---->
      <!----><span class=\\"v-card__underlay\\"></span>
    </div>
  </div>
</div>"
`;

exports[`HomeView > IDIR User 1`] = `
"<div>
  <div class=\\"v-container v-locale--is-ltr\\">
    <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
      <!---->
      <div class=\\"v-card__loader\\">
        <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
          <!---->
          <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
          <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
            <div class=\\"v-progress-linear__indeterminate\\">
              <div class=\\"v-progress-linear__indeterminate long\\"></div>
              <div class=\\"v-progress-linear__indeterminate short\\"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div class=\\"v-card-title\\">Ministry User</div>
      <div class=\\"v-card-text\\">
        <div class=\\"v-list v-theme--light v-list--density-default v-list--one-line\\" tabindex=\\"0\\" role=\\"listbox\\">
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">IDIR:</div>
              <div class=\\"v-list-item-subtitle\\">idiruser</div>
            </div>
            <!---->
          </div>
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">GUID:</div>
              <div class=\\"v-list-item-subtitle\\"></div>
            </div>
            <!---->
          </div>
          <!--v-if-->
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">Name:</div>
              <div class=\\"v-list-item-subtitle\\"> </div>
            </div>
            <!---->
          </div>
          <div class=\\"v-list-item v-theme--light v-list-item--density-default v-list-item--one-line v-list-item--variant-text\\">
            <!----><span class=\\"v-list-item__underlay\\"></span>
            <!---->
            <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
              <!---->
              <!---->
              <div class=\\"v-list-item-title\\">Email:</div>
              <div class=\\"v-list-item-subtitle\\"></div>
            </div>
            <!---->
          </div>
        </div>
      </div>
      <!---->
      <!----><span class=\\"v-card__underlay\\"></span>
    </div>
  </div>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>"
`;
